bigquery_etl/view/__init__.py

```diff
@@ -176,27 +176,27 @@ def udf_references(self):
 
     @cached_property
     def view_schema(self):
-        """Derive view schema from a schema file or a dry run result."""
-        schema_file = Path(self.path).parent / "schema.yaml"
+        """Derive view schema from a dry run result and/or a schema file."""
+        schema = None
+
+        # check schema based on dry run results
```

Comment:
I think dry running everything makes sense to make sure everything is updated but it will probably add a few minutes to the publish views task.  Views should usually be up to date since the generator already ran.  I don't have any issues with this but worth checking later on to see how much time this adds.

Reply:
I did manage to get the dry-runs on some of the views that select from large tables to run quicker by adding partition column filters, but I'm still going to leave this with the original behavior of using the schema from `schema.yaml` by default, with the `derived_view_schemas` SQL generator using custom logic to always try to dry-run the views.